messaging/src/main/java/org/axonframework/tracing/TracingHandlerEnhancerDefinition.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) 2010-2022. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.tracing;
+
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.HandlerEnhancerDefinition;
+import org.axonframework.messaging.annotation.MessageHandlingMember;
+import org.axonframework.messaging.annotation.WrappedMessageHandlingMember;
+
+import java.lang.reflect.Executable;
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import javax.annotation.Nonnull;
+
+/**
+ * Enhances message handlers with the provided {@link SpanFactory}, wrapping handling of the message in a {@link Span}
+ * that is reported to the monitoring tooling.
+ * <p>
+ * Since {@code EventSourcingHandlers} can be very noisy when loading the aggregate, they can be enabled or disabled
+ * separately through constructor configuration.
+ *
+ * @author Mitchell Herrijgers
+ * @since 4.6.0
+ */
+public class TracingHandlerEnhancerDefinition implements HandlerEnhancerDefinition {
+
+    private final SpanFactory spanFactory;
+    private final boolean showEventSourcingHandlers;
+
+    /**
+     * Creates a new {@link TracingHandlerEnhancerDefinition}, adding spans to each message handler invocation.
+     *
+     * @param spanFactory               The {@link SpanFactory} to use.
+     * @param showEventSourcingHandlers Whether to show event sourcing handlers as a span. Can be very noisy when
+     *                                  loading large aggregates.
+     */
+    public TracingHandlerEnhancerDefinition(SpanFactory spanFactory, boolean showEventSourcingHandlers) {
```

Comment:
Might fell a little redundant given the size, but why not add a Builder setup here too? That way, we'll be free to remove/add/adjust parameters and their defaults, without constructing an abundance of constructors.

I'll leave the yes/no to do this to you. 
My gut tells me this class might change soon(ish) because of the role it fulfills.
To me, that merits the introduction of the builder paradigm.
If you think otherwise, I trust in your judgement.

Reply:
No you are corrrect, changing it to a Builder